nvdaHelper/UIARemote/remoteLog.h

```diff
@@ -29,27 +29,39 @@ class RemoteableLogger {
 		scope.BindResult(_log);
 	}
 
-	template<typename T> RemoteableLogger& operator <<(T& message) {
-		if constexpr(std::is_base_of<UiaInt,T>::value) {
-			_log.Append(message.Stringify());
-		} else {
-			_log.Append(message);
-		}
+	RemoteableLogger& operator <<(UiaInt& message) {
+		_log.Append(message.Stringify());
+		return *this;
+	}
+
+	RemoteableLogger& operator <<(UiaString& message) {
+		_log.Append(message);
+		return *this;
+	}
+
+	RemoteableLogger& operator <<(const std::wstring message) {
+		_log.Append(message);
 		return *this;
 	}
 
 	void dumpLog() {
 		assert(!UiaOperationAbstraction::ShouldUseRemoteApi());
 		std::wstring messageBlock{L"Dump log start:\n"};
 		try {
-			auto v = *_log;
+			// locally, a UiaArray is a shared_ptr to a vector of will_shared_bstr 
+			const std::vector<wil::shared_bstr>& v = *_log;
 			for(const auto& message: v) {
 				messageBlock+=message.get();
 			}
 		} catch (std::exception& e) {
-			std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
-			auto what = converter.from_bytes(e.what());
-			messageBlock+=L"dumpLog exception: " + what + L"\n";
+			const std::string what{e.what()};
+			// convert exception message to unicode
+			int wideLen = MultiByteToWideChar(CP_UTF8, 0, what.c_str(), what.length(), nullptr, 0);
+			auto wideBuf = std::make_unique<wchar_t[]>(wideLen);
+			MultiByteToWideChar(CP_UTF8, 0, what.c_str(), what.length(), wideBuf.get(), wideLen);
+			messageBlock += L"dumpLog exception: ";
+			messageBlock += wideBuf.get();
+			messageBlock += L"\n";

```

Comment:
See other exception handling comments. I'd extract this out into a shared utility.